src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryMock.java

```diff
@@ -703,9 +705,17 @@ public boolean isEmpty()
 	 * @return An inventory snapshot.
 	 */
 	@NotNull
+	@ApiStatus.Internal
 	public Inventory getSnapshot()
 	{
-		return new InventoryMock(this);
+		try
+		{
+			return this.getClass().getConstructor(this.getClass()).newInstance(this);
+		}
+		catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException e)
+		{
+			throw new IllegalStateException(e);
+		}
```

Comment:
Alternative suggestion for this to avoid using the reflection:
```java
		if (InventoryMock.class.equals(getClass()))
		{
			return new InventoryMock(this);
		}
		else
		{
			throw new UnimplementedOperationException(String.format("%s does not implement method getSnapshot", this.getClass().getSimpleName()));
		}
```
At the moment all the others classes implement this, but if an new one is added, this would notify us about that.